jaeger_client/metrics.py

```diff
@@ -21,69 +21,122 @@
 from __future__ import absolute_import
 
 
-class Metrics(object):
-    """Provides an abstraction of metrics reporting framework."""
+class MetricsFactory(object):
+    """Generates new metrics."""
 
-    # TODO do we want to automatically include host name?
-    prefix = 'jaeger'
+    def __init__(self, tags=None):
+        self._tags = tags
 
-    TRACES_STARTED_SAMPLED = '%s.traces-started.sampled' % prefix
-    """Number of traces started by this tracer as sampled"""
+    def _noop(self, *args):
+        pass
 
-    TRACES_STARTED_NOT_SAMPLED = '%s.traces-started.not-sampled' % prefix
-    """Number of traces started by this tracer as not sampled"""
+    def create_counter(self, name, tags=None):
+        """
+        Generates a new counter from the given name and tags and returns
+        a callable function used to increment the counter.
+        :param name: name of the counter
+        :param tags: tags for the counter
+        :return: a callable function which takes the value to increase
+        the counter by ie. def increment(value)
+        """
+        return self._noop
 
-    TRACES_JOINED_SAMPLED = '%s.traces-joined.sampled' % prefix
-    """Number of externally started sampled traces this tracer joined"""
+    def create_timer(self, name, tags=None):
+        """
+        Generates a new timer from the given name and tags and returns
+        a callable function used to record a duration.
+        :param name: name of the timer
+        :param tags: tags for the timer
+        :return: a callable function which takes the duration to
+        record ie. def record(duration)
```

Comment:
so clay uses milliseconds and m3 used microseconds, ill make it microseconds and cast it for clay (legacy)